### ISSUE TITLE
NTXentLoss for Binary Task

### ISSUE THREAD

**Issue by icsd13152** [KevinMusgrave/pytorch-metric-learning#579]
Hi @KevinMusgrave,

How can I use the NTXentLoss for a binary Task where my classes are 0 or 1?
I know that based on similarity matrix the main diagonal is the score distribution for element wise in batch for the positive examples.
The other scores are the negatives. 
But How can I call your function in order to have a True or False (0 or 1) as target label and ground truth?

Could you pelase exlain?

Thanks

**Comment by KevinMusgrave** [comment]
Here's an example:

```python
# batch of 8 embeddings
embeddings = torch.randn(8, 128)
# a random set of 0/1 labels
labels = torch.tensor([0, 0, 1, 0, 1, 1, 1, 0])

loss = loss_fn(embeddings, labels)
```

**Comment by icsd13152** [comment]
Hi,

If I unterstand correctly, the embeddings is a tupple of pairs? or in the implementation of this loss all of the cosine similarities are calculated between all of the possible pairs in batch?
In other words in index 2 of Tensor Labels we have ground truth equals to 1. In the same index at embeddings Tensor we have a tuple of a positive pair (i.e (emb1,emb2) )?

Thanks

**Comment by KevinMusgrave** [comment]
No, embeddings is of size (N,D) where N is the batch size and D is the embedding dimensionality.

`label[2] = 1` means that `embeddings[2]` has a label of 1. Positive pairs are formed by embeddings with the same label.

**Comment by KevinMusgrave** [comment]
Some relevant links:

- https://github.com/KevinMusgrave/pytorch-metric-learning/issues/6#issuecomment-1385678799
- https://github.com/KevinMusgrave/pytorch-metric-learning/issues/549
